File: src/main/java/com/sumup/purecode/users/UserDto.java
package com.sumup.purecode.users;

public record UserDto(Long id, String firstName, String lastName, String email) {
    static UserDto fromEntity(UserEntity entity) {
        return new UserDto(
                entity.getId(),
                entity.getFirstName(),
                entity.getLastName(),
                entity.getEmail()
        );
    }
}

File: src/main/java/com/sumup/purecode/users/UserDto.kt
package com.sumup.purecode.users

data class UserDto(val id: Long, val firstName: String, val lastName: String, val email: String) {

    companion object {
        fun fromEntity(userEntity: UserEntity): UserDto = with(userEntity) {
            UserDto(id!!, firstName!!, lastName!!, email!!)
        }
    }
}
